src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -337,6 +337,11 @@ private void ProcessExpressionParameter()
             if (ExcludeProperty != null)
             {
                 _exclusionFilter = new MshExpressionFilter(ExcludeProperty);
+                if ((Property == null) || (Property.Length == 0))
+                {
```

Comment:
I analyzed the usages of `Select-Object` in GitHub to find the usages like `Select-Object -ExcludeProperty ProcessName -ExpandProperty Modules`, which might be broken by this fix. The analysis scripts and results can be found [here](https://gist.github.com/daxian-dbw/07847ff5ef6da2c100e8b22bd94f0feb).

1074 .ps1/.psm1 files were found in GitHub with `Select-Object` in them, and totally 3376 instances of `Select-Object` were found. Out of them, 2 usages might be broken by the fix on `-ExcludeProperty`:
```
## FIND 2 usages like 'Select-Object -ExcludeProperty value -ExpandProperty value'
PS:96> $results | ? BreakingCase

UsesSelectObject     : True
FileName             : F:\temp\selectobject-dir-psm1\ECSNamespace.psm1
NeedManualInspection : False
BreakingCase         : True
Line                 : 48

UsesSelectObject     : True
FileName             : F:\temp\selectobject-dir-psm1\ECSvdc.psm1
NeedManualInspection : False
BreakingCase         : True
Line                 : 158
```
[ECSNamespace.psm1](https://raw.githubusercontent.com/bottkars/ECShell/6b2531a84e5107ccb1c6ab71a18e766f057efa14/Multitenancy/ECSNamespace.psm1) has `Select-Object -ExpandProperty $Expandproperty -ExcludeProperty $Excludeproperties` in it:
```
Invoke-RestMethod -Uri $Uri -Headers $ECSAuthHeaders -Method Get  -ContentType $ContentType | Select-Object -ExpandProperty $Expandproperty -ExcludeProperty $Excludeproperties | Select-Object $Properties #,@{N="NamespaceID";E={$_.id}}
```
[ECSvdc.psm1](https://raw.githubusercontent.com/bottkars/ECShell/6b2531a84e5107ccb1c6ab71a18e766f057efa14/Provisioning/ECSvdc.psm1) has `Select-Object -ExpandProperty vdc -ExcludeProperty $Excludeproperty` in it:
```
Invoke-RestMethod -Uri $Uri -Headers $ECSAuthHeaders -Method $Method -ContentType $ContentType  | Select-Object -ExpandProperty $Expandproperty | Select-Object -ExpandProperty vdc -ExcludeProperty $Excludeproperty
```

Although we actually find the usages that might be broken by this fix, but those 2 usages seems wrong in the first place -- the author indented to exclude some properties, which actually doesn't happen. 

Given the data, I feel it's OK to have this "breaking change" introduced by this fix.